tests/Unit/Domain/Services/TalkRating/OneToTenRatingTest.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Copyright (c) 2013-2017 OpenCFP
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ *
+ * @see https://github.com/opencfp/opencfp
+ */
+
+namespace OpenCFP\Test\Unit\Domain\Services\TalkRating;
+
+use Mockery;
+use Mockery\Adapter\Phpunit\MockeryPHPUnitIntegration;
+use OpenCFP\Domain\Model\TalkMeta;
+use OpenCFP\Domain\Services\Authentication;
+use OpenCFP\Domain\Services\TalkRating\OneToTenRating;
+
+/**
+ * @covers \OpenCFP\Domain\Services\TalkRating\OneToTenRating
+ */
+class OneToTenRatingTest extends \PHPUnit\Framework\TestCase
+{
+    use MockeryPHPUnitIntegration;
+
+    /**
+     * @dataProvider validRatingProvider
+     */
+    public function testGoodRatingsAreSuccessful($rating)
+    {
+        $mockAuth    = Mockery::mock(Authentication::class)->shouldIgnoreMissing();
+        $metaMock    = Mockery::mock(TalkMeta::class);
+        $oneToTen    = new OneToTenRating($metaMock, $mockAuth);
+        $this->assertTrue($oneToTen->isValidRating($rating));
+    }
+
+    /**
+     * @dataProvider invalidRatingProvider
+     */
+    public function testBadRatingsAreNotSuccessFul($rating)
+    {
+        $mockAuth    = Mockery::mock(Authentication::class)->shouldIgnoreMissing();
+        $metaMock    = Mockery::mock(TalkMeta::class);
+        $oneToTen    = new OneToTenRating($metaMock, $mockAuth);
+        $this->assertFalse($oneToTen->isValidRating($rating));
+    }
+
+    public function testGetRatingNameReturnsOneToTen()
+    {
+        $mockAuth = Mockery::mock(Authentication::class);
+        $mockAuth->shouldReceive('userId');
+        $metaMock    = Mockery::mock(TalkMeta::class);
+        $oneToTen    = new OneToTenRating($metaMock, $mockAuth);
+        $this->assertSame('OneToTen', $oneToTen->getRatingName());
+    }
+
+    public function validRatingProvider(): array
+    {
+        return [
+            [0],
```

Comment:
If we call the rating strategy `OneToTenRating`, shouldn't `0` be out of range?

What do you think?

Reply:
Zero kind of functions as a 'deleted' rating.
Say i was reading a talk, then somehow gave it a rating of 8 by accident. Now i didn't want to rate this talk yet, so i just remove the current rating, which in this system sets it to 0.

That, or the logic has to be changed around to delete the rating from the database when a certain value is given.